### DISCUSSION TITLE
Is it possible to use the Javascript activity to read data from context variable and insert into Application database?

### DISCUSSION THREAD

**Discussion by jayachandra21** [elsa-workflows/elsa-core#1636] [Q&A]
Hi @sfmskywalker ,

One question with regarding to the JavaScript activity that, I have an application data in the  workflow context variable.
So I need to insert the data into the application database. So does Javascript activity support this requirement?
If possible, please share any reference material.

**Comment by sfmskywalker** [comment]
You could extend the JavaScript language with custom functions that allow you to update your database, but a small downside is that the JavaScript engine does not currently support `async/await` operations.

If that's not a big issue for you, [this page](https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-javascript) explains how to add custom JS functions.

Alternatively, since you mentioned the `workflowContext` variable, you could take advantage of the workflow context API services to automatically persist your object.

The way this works is that you implement a workflow context provider class that handles loading as well as saving workflow contexts (which represents your DB entity to be loaded/persisted).

More information can be found [here](https://elsa-workflows.github.io/elsa-core/docs/next/concepts/concepts-workflow-context).

Finally, you could implement a custom activity that allows you to save the context variable into the database explicitly. This of course does support `async/await` operations.
